code/obj/item/mob_parts/artifact_parts.dm

```diff
@@ -0,0 +1,540 @@
+											/datum/ailment/disease/robotic_transformation, /datum/ailment/disease/corrupt_robotic_transformation,
+											/datum/ailment/disease/good_robotic_transformation, /datum/ailment/disease/noheart,
+											/datum/ailment/disease/kidney_failure/left, /datum/ailment/disease/kidney_failure/right,
+											/datum/ailment/disease/liver_failure, /datum/ailment/disease/respiratory_failure/left,
+											/datum/ailment/disease/respiratory_failure/right,
+											/datum/ailment/parasite/headspider)
+
+	New(datum/abilityHolder/holder)
+		..()
+		if (src.allowed_ailments)
+			return
+		src.allowed_ailments = list()
+		for (var/ailment in typesof(/datum/ailment))
+			src.allowed_ailments[ailment] = TRUE
+		for (var/ailment in typesof(/datum/ailment/malady))
+			src.allowed_ailments[ailment] = FALSE
+		for (var/ailment in unhealable_ailments)
+			src.allowed_ailments[ailment] = FALSE
+
+	cast(atom/target)
+		if (get_dist(holder.owner, target) > 1)
+			boutput(holder.owner, "<span class='alert'>The target is too far away!</span>")
+			return TRUE
+		if (!isliving(target) || issilicon(target) || isintangible(target) || islivingobject(target))
+			boutput(holder.owner, "<span class='alert'>Your power cannot be used on this target!</span>")
+			return TRUE
+		if (target == holder.owner)
+			boutput(holder.owner, "<span class='alert'>It could be dangerous using this power on yourself!</span>")
+			return TRUE
+		if (is_incapacitated(holder.owner))
+			boutput(holder.owner, "<span class='alert'>You can't use your power while incapacitated!</span>")
+			return TRUE
+		if (holder.owner.z == Z_LEVEL_NULL)
+			return TRUE
+
+		var/mob/living/M = target
+
+		if (length(M.ailments))
+			if (M.cure_disease_by_path(/datum/ailment/malady/flatline))
+				boutput(holder.owner, "<span class='alert'>You heal [M]'s heart attack!</span>")
+				boutput(M, "<span class='notice'>The pain in your heart suddenly goes away!</span>")
+				return
+
+			for (var/datum/ailment_data/A as anything in M.ailments)
+				if (A.master?.type && src.allowed_ailments[A.master.type] && M.cure_disease_by_path(A.master.type))
+					boutput(holder.owner, "<span class='alert'>You heal [M] of an ailment!</span>")
+					boutput(M, "<span class='notice'>You suddenly feel a lot better!</span>")
+					return
+
+		if (length(M.implant) && (locate(/obj/item/implant/projectile) in M.implant))
+			for (var/obj/item/implant/projectile/I in M.implant)
+				I.on_remove(M)
+				M.implant.Remove(I)
+				I.set_loc(M.loc)
+				// scatter on ground
+				I.pixel_x = rand(-11, 8)
+				I.pixel_y = rand(-9, -5)
+			boutput(holder.owner, "<span class='alert'>You pull foreign material out of [M]!</span>")
+			boutput(M, "<span class='notice'>You feel some bad stuff fall out of your chest!</span>")
+			return
+
+		boutput(holder.owner, "<span class='alert'>[M] has nothing wrong with them that can be healed!</span>")
```

Comment:
"by your hands"? "by your touch"?

Reply:
Would be cool but idk feels like it stretches the sentence a bit?